### ISSUE TITLE
Clear secrets when extension is uninstalled

### ISSUE THREAD

**Issue by RMacfarlane** [microsoft/vscode#123817]
I am relying on the fact that secrets are not cleared after every session. I am using it to store PAT token.

I think the secrets are also not cleared when extensions are uninstalled. I uninstalled my extension, reloaded and reinstalled the extension and the secrets still existed. It would be helpful if uninstalling extension will clear out the secrets. Which brings me to a potential feature to force clear all secrets of an extension similar to how browser devtools provide a way to force clear cache and application data. Will be again helpful in situations where extension authors do not provide a way to clear the secrets.

Let me know if these needs to go to separate issue as this is about the documentation.

_Originally posted by @ankitbko in https://github.com/microsoft/vscode/issues/115252#issuecomment-773886577_

**Comment by DumpySquare** [comment]
+1 to this.  feature parity with keytar would be nice.  

- set specific (keyed) key rings
- list keys (can be used to loop through and clear the ring)

But would settle for a clear function, or at minimum, have the key ring cleared when extension is uninstalled.

**Comment by TylerLeonhardt** [comment]
@DumpySquare we wouldn't allow you to clear secrets of another extension... however the original ask to clear when an extension is uninstalled is something I really wanna do.

**Comment by DumpySquare** [comment]
@TylerLeonhardt, agreed.  I don't need to do anything for other extensions, but if each keytar instance was "keyed" for each extension, then you could have a function to clear the keychain as needed.  

To me, I would like to see all native keytar functions exposed in vscode, but then I start to question the value of having it integrated.  For now, regular keytar does what I need, but I guess it would be nice to have less dependencies down the road with an integrated solution like this.

**Comment by TylerLeonhardt** [comment]
@DumpySquare can you get a new issue going with what kind of API you'd like? What do you think of SecretStorage having a `keys()` function that gets all the keys (aka accounts) in keytar.

![image](https://user-images.githubusercontent.com/2644648/145110594-8a911e76-e90c-463a-bb6b-1a5fded0590a.png)

**Comment by TylerLeonhardt** [comment]
@DumpySquare related: https://github.com/microsoft/vscode/issues/115215

**Comment by DumpySquare** [comment]
@TylerLeonhardt, I'm not totally following your keys() function suggestion, but having full feature paraty of keytar exposed within vscode would be the goal.  The current secret storage is a great start and I'm sure it helped for many use cases our there.  

The following is how we use keytar.  Being able to set the keyring name "f5Hosts" allows me to set different "sets" of keys for different purposes.  Even though we only use one keyring, it allows me to scope how I keep the secrets, very much like github env/secretes.  This limited scope also works well for listing all the keys and/or deleting/clearing entire sets as needed.

https://github.com/f5devcentral/vscode-f5/blob/aca728d8b150ff7c064eade5b68433a5f4ce9a8c/src/deviceImport.ts#L182

Thanks for linking that other issue.  I guess I need more information about where vscode is going with this and what it means to how vscode-f5 extension uses keytar.  For now, the external keytar project does what we need.

**Comment by DumpySquare** [comment]
@TylerLeonhardt any updates on this issue?  The .keys() function would be a solution for me.  Do you need another issue to provide the .keys() function or can we just utilize this issue?

**Comment by TylerLeonhardt** [comment]
@DumpySquare another issue would be nice.

**Comment by lppedd** [comment]
Gentle ping. This would be nice to have.
